rllib/algorithms/dqn/dqn.py

```diff
@@ -374,6 +456,141 @@ def training_step(self) -> ResultDict:
         Returns:
             The results dict from executing the training iteration.
         """
+        # New API stack (RLModule, Learner, EnvRunner, ConnectorV2).
+        if self.config.uses_new_env_runners:
+            return self._training_step_new_api_stack()
+        # Old and hybrid API stacks (Policy, RolloutWorker, Connector, maybe RLModule,
+        # maybe Learner).
+        else:
+            return self._training_step_old_and_hybrid_api_stack()
+
+    def _training_step_new_api_stack(self) -> ResultDict:
+        # Alternate between storing and sampling and training.
+        store_weight, sample_and_train_weight = calculate_rr_weights(self.config)
+        train_results = {}
+
+        # Run multiple sampling iterations.
+        for _ in range(store_weight):
+            with self._timers[SAMPLE_TIMER]:
+                # TODO (simon): Use `sychnronous_parallel_sample()` here.
```

Comment:
We continue kicking this can down the road :D 
Can we do this in this PR, fix the `sychnronous_parallel_sample` function? I think it's really just a few lines that would have to be changed in there to make it work with episodes.

Then we can also - in this same PR - fix SAC and PPO for good and remove all these TODOs.

Reply:
One merge away :)